@here/harp-mapview/lib/text/TextElementState.ts

```diff
@@ -206,28 +203,37 @@ export class TextElementState {
     }
 
     /**
-     * @param viewDistance Current distance of the element to the view center.
+     * Initialize text and icon render states
      */
-    private initialize(viewDistance: number) {
+    private initializeRenderStates() {
         assert(this.m_textRenderState === undefined);
         assert(this.m_iconRenderStates === undefined);
 
-        this.setViewDistance(viewDistance);
-
+        const { textFadeTime } = this.element;
+        const iconFadeTime = this.element.poiInfo?.technique.iconFadeTime;
         if (this.element.type === TextElementType.LineMarker) {
             this.m_iconRenderStates = new Array<RenderState>();
             for (const _point of this.element.points as THREE.Vector3[]) {
                 const iconRenderStates = this.m_iconRenderStates as RenderState[];
                 const renderState = new RenderState();
+                if (iconFadeTime !== undefined) {
+                    renderState.fadeTime = iconFadeTime;
+                }
```

Comment:
We should be able to do:
`const renderState = new RenderState(iconFadeTime);`

If undefined, it'll take the default parameter value. Right now we have 3 other parameters before it, which are never used (except for a couple of test here). I would move the fade time parameter to the first position and even remove the other constructor parameters. I don't see the use case of initializing the state with custom values for startTime,opacity etc...